[linux] undefined reference to `launch_activate_socket'

When running `go run main.go` it crashes with the following error:
```
# github.com/moomernman/zap/vendor/github.com/puma/puma-dev/dev/launch
/tmp/go-build564753384/b129/_x002.o: In function `_cgo_f28cf972db4f_Cfunc_launch_activate_socket':
/tmp/go-build/cgo-gcc-prolog:55: undefined reference to `launch_activate_socket'
collect2: error: ld returned 1 exit status
```

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -21,20 +21,23 @@
   revision = "ae77be60afb1dcacde03767a8c37337fad28ac14"
 
 [[projects]]
-  digest = "1:463e4140189f8194f9121ca1c7fe3b8e9e9a2ab3d949b43c835c21034927dc62"
+  digest = "1:d64a0aa67f715fd28f5ddfe0d74c06b6d6fe0fed931d77e662f2c4694f6854dc"
   name = "github.com/miekg/dns"
   packages = ["."]
   pruneopts = "UT"
-  revision = "5a2b9fab83ff0f8bfc99684bd5f43a37abe560f1"
-  version = "v1.0.8"
+  revision = "56be65265e34e731425e0269a301774938827c60"
+  version = "v1.1.3"
 
 [[projects]]
   branch = "master"
-  digest = "1:94602f5f1e0be5db549f45ca29a39143118ce1a1971c9088d3578bf12cd7fb4a"
+  digest = "1:f515c83dabdd006b4b782b5252e1c510630b1e75b3799c180b4516079ba4caee"
   name = "github.com/moomerman/go-lib"
-  packages = ["rproxy"]
+  packages = [
+    "rproxy",
+    "rproxy/wsutil",
+  ]
   pruneopts = "UT"
-  revision = "2df34aab356ec4abb6f6d38ab780377101c4857a"
+  revision = "6b20bf86751ae55debd5c76698227e2421f616c8"
 
 [[projects]]
   digest = "1:07a53954325f093c73561db73561cb728efeadb8bc3b4a6a989acb1391a6b51d"
@@ -50,11 +53,11 @@
 
 [[projects]]
   branch = "master"
-  digest = "1:f099f01404787f3d89257d3dd7fbea0c8da7bd9fd3ff9ae9d8bf80c56523b125"
+  digest = "1:d81a708124c38e074fd655f8e9e9218e9461eb208cf23d787b58b193fd3547af"
   name = "github.com/unrolled/render"
   packages = ["."]
   pruneopts = "UT"
-  revision = "4206df6ff70175d219fdb4672ae157fbed820103"
+  revision = "449f3985007486f7f42ca7f62b4c8b45d6d89cbf"
 
 [[projects]]
   branch = "master"
@@ -66,14 +69,6 @@
 
 [[projects]]
   branch = "master"
-  digest = "1:f7a72935dedbd57d651ea4870078bfb61a5c1728c5ed0b3058781e558df0f8a3"
-  name = "github.com/yhat/wsutil"
-  packages = ["."]
-  pruneopts = "UT"
-  revision = "1d66fa95c997864ba4d8479f56609620fe542928"
-
-[[projects]]
-  branch = "master"
   digest = "1:d5891c5bca9c62e5d394ca26491d2b710a1dc08cedeb0ca8f9ac4c3305120b02"
   name = "golang.org/x/crypto"
   packages = [
@@ -81,11 +76,11 @@
     "ed25519/internal/edwards25519",
   ]
   pruneopts = "UT"
-  revision = "0e37d006457bf46f9e6692014ba72ef82c33022c"
+  revision = "64072686203f69e3fd20143576b27200f18ab0fa"
 
 [[projects]]
   branch = "master"
-  digest = "1:9b6ce01ec663e8cccfab3c665d3f9fcea5f7e74fe473ea18bce86008214234f9"
+  digest = "1:590b253cf66508d019ea03cac62a6156d12c0ca1869242f31c5362eac67f1632"
   name = "golang.org/x/net"
   packages = [
     "bpf",
@@ -99,7 +94,15 @@
     "ipv6",
   ]
   pruneopts = "UT"
-  revision = "26e67e76b6c3f6ce91f7c52def5af501b4e0f3a2"
+  revision = "ed066c81e75eba56dd9bd2139ade88125b855585"
+
+[[projects]]
+  branch = "master"
+  digest = "1:64ec1e1de5cec5e1579df2ff7ac10b20e3bf1cb4393846f631fb204dd9cb44f6"
+  name = "golang.org/x/sys"
+  packages = ["unix"]
+  pruneopts = "UT"
+  revision = "054c452bb702e465e95ce8e7a3d9a6cf0cd1188d"
 
 [[projects]]
   digest = "1:a2ab62866c75542dd18d2b069fec854577a20211d7c0ea6ae746072a1dccdd18"

--- a/cert/cert_linux.go
+++ b/cert/cert_linux.go
@@ -7,48 +7,6 @@ const supportDir = "~/.zap/ssl"
 // InstallCert installs a CA certificate root in the system cacerts on linux
 func InstallCert(cert string) error {
 	fmt.Printf("! Add %s to your browser to trust CA\n", cert)
+	fmt.Printf("* See https://github.com/moomerman/zap/wiki/Linux")
 	return nil
-
-	// determine if we're running Ubuntu (and what version) with
-	// lsb_release -a
-	// cat /etc/*release
 }
-
-// https://thomas-leister.de/en/how-to-import-ca-root-certificate/
-
-// #!/bin/bash
-
-// ### Script installs root.cert.pem to certificate trust store of applications using NSS
-// ### (e.g. Firefox, Thunderbird, Chromium)
-// ### Mozilla uses cert8, Chromium and Chrome use cert9
-
-// ###
-// ### Requirement: apt install libnss3-tools
-// ###
-
-// ###
-// ### CA file to install (CUSTOMIZE!)
-// ###
-
-// certfile="cert.pem"
-// certname="zap CA"
-
-// ###
-// ### For cert8 (legacy - DBM)
-// ###
-
-// for certDB in $(find ~/ -name "cert8.db")
-// do
-//     certdir=$(dirname ${certDB});
-//     certutil -A -n "${certname}" -t "TCu,Cu,Tu" -i ${certfile} -d dbm:${certdir}
-// done
-
-// ###
-// ### For cert9 (SQL)
-// ###
-
-// for certDB in $(find ~/ -name "cert9.db")
-// do
-//     certdir=$(dirname ${certDB});
-//     certutil -A -n "${certname}" -t "TCu,Cu,Tu" -i ${certfile} -d sql:${certdir}
-// done

--- a/cert/cert_linux.go
+++ b/cert/cert_linux.go
@@ -2,7 +2,7 @@ package cert
 
 import "fmt"
 
-const supportDir = "~/.puma-dev-ssl"
+const supportDir = "~/.zap/ssl"
 
 // InstallCert installs a CA certificate root in the system cacerts on linux
 func InstallCert(cert string) error {
@@ -13,3 +13,42 @@ func InstallCert(cert string) error {
 	// lsb_release -a
 	// cat /etc/*release
 }
+
+// https://thomas-leister.de/en/how-to-import-ca-root-certificate/
+
+// #!/bin/bash
+
+// ### Script installs root.cert.pem to certificate trust store of applications using NSS
+// ### (e.g. Firefox, Thunderbird, Chromium)
+// ### Mozilla uses cert8, Chromium and Chrome use cert9
+
+// ###
+// ### Requirement: apt install libnss3-tools
+// ###
+
+// ###
+// ### CA file to install (CUSTOMIZE!)
+// ###
+
+// certfile="cert.pem"
+// certname="ZAP CA"
+
+// ###
+// ### For cert8 (legacy - DBM)
+// ###
+
+// for certDB in $(find ~/ -name "cert8.db")
+// do
+//     certdir=$(dirname ${certDB});
+//     certutil -A -n "${certname}" -t "TCu,Cu,Tu" -i ${certfile} -d dbm:${certdir}
+// done
+
+// ###
+// ### For cert9 (SQL)
+// ###
+
+// for certDB in $(find ~/ -name "cert9.db")
+// do
+//     certdir=$(dirname ${certDB});
+//     certutil -A -n "${certname}" -t "TCu,Cu,Tu" -i ${certfile} -d sql:${certdir}
+// done

--- a/cert/cert_linux.go
+++ b/cert/cert_linux.go
@@ -31,7 +31,7 @@ func InstallCert(cert string) error {
 // ###
 
 // certfile="cert.pem"
-// certname="ZAP CA"
+// certname="zap CA"
 
 // ###
 // ### For cert8 (legacy - DBM)

--- a/cert/cert_windows.go
+++ b/cert/cert_windows.go
@@ -4,12 +4,9 @@ import "fmt"
 
 const supportDir = "~/.zap/ssl"
 
-// InstallCert installs a CA certificate root in the system cacerts on linux
+// InstallCert installs a CA certificate root in the system cacerts on windows
 func InstallCert(cert string) error {
 	fmt.Printf("! Add %s to your browser to trust CA\n", cert)
+	fmt.Printf("* See https://github.com/moomerman/zap/wiki/Windows")
 	return nil
-
-	// determine if we're running Ubuntu (and what version) with
-	// lsb_release -a
-	// cat /etc/*release
 }

--- a/cert/cert_windows.go
+++ b/cert/cert_windows.go
@@ -1,0 +1,15 @@
+package cert
+
+import "fmt"
+
+const supportDir = "~/.zap/ssl"
+
+// InstallCert installs a CA certificate root in the system cacerts on linux
+func InstallCert(cert string) error {
+	fmt.Printf("! Add %s to your browser to trust CA\n", cert)
+	return nil
+
+	// determine if we're running Ubuntu (and what version) with
+	// lsb_release -a
+	// cat /etc/*release
+}

--- a/dns/example_test.go
+++ b/dns/example_test.go
@@ -21,5 +21,5 @@ func Example() {
 	dns.Address = fmt.Sprintf("127.0.0.1:%d", port)
 	log.Println("* DNSServer", dns.Address)
 
-	dns.Serve(domains)
+	dns.Serve()
 }

--- a/zap/install_linux.go
+++ b/zap/install_linux.go
@@ -7,3 +7,9 @@ func installService(httpAddr, httpsAddr string) error {
 func uninstallService() error {
 	return nil
 }
+
+// sudo setcap 'cap_net_bind_service=+ep' zapd
+// ./zapd -dns 127.0.0.54:53
+
+// /etc/systemd/resolved.conf
+// DNS=127.0.0.54

--- a/zap/install_linux.go
+++ b/zap/install_linux.go
@@ -7,9 +7,3 @@ func installService(httpAddr, httpsAddr string) error {
 func uninstallService() error {
 	return nil
 }
-
-// sudo setcap 'cap_net_bind_service=+ep' zapd
-// ./zapd -dns 127.0.0.54:53
-
-// /etc/systemd/resolved.conf
-// DNS=127.0.0.54

--- a/zap/install_linux.go
+++ b/zap/install_linux.go
@@ -1,0 +1,9 @@
+package zap
+
+func installService(httpAddr, httpsAddr string) error {
+	return nil
+}
+
+func uninstallService() error {
+	return nil
+}

--- a/zap/install_windows.go
+++ b/zap/install_windows.go
@@ -1,0 +1,12 @@
+package zap
+
+func installService(httpAddr, httpsAddr string) error {
+	return nil
+}
+
+func uninstallService() error {
+	return nil
+}
+
+// symbolic links via powershell (requires admin)
+// New-Item -path ~\.zap\mysite.test -itemType SymbolicLink -target ~\Workspace\mysite

--- a/zap/install_windows.go
+++ b/zap/install_windows.go
@@ -7,6 +7,3 @@ func installService(httpAddr, httpsAddr string) error {
 func uninstallService() error {
 	return nil
 }
-
-// symbolic links via powershell (requires admin)
-// New-Item -path ~\.zap\mysite.test -itemType SymbolicLink -target ~\Workspace\mysite

--- a/zap/server.go
+++ b/zap/server.go
@@ -4,13 +4,11 @@ import (
 	"context"
 	"crypto/tls"
 	"log"
-	"net"
 	"net/http"
 	"sync"
 	"time"
 
 	"github.com/moomerman/zap/cert"
-	"github.com/puma/puma-dev/dev/launch"
 	"golang.org/x/net/http2"
 )
 
@@ -57,40 +55,6 @@ func (s *Server) Stop() {
 	s.https.Shutdown(ctx)
 }
 
-func (s *Server) serveHTTP() error {
-	var listener net.Listener
-	var err error
-
-	if s.HTTPAddr == "Socket" {
-		listener = getSocketListener(s.HTTPAddr)
-	} else {
-		listener, err = net.Listen("tcp", s.HTTPAddr)
-		if err != nil {
-			log.Fatal("[zap] unable to create listener", err)
-		}
-	}
-
-	log.Println("[zap] http listening at", listener.Addr())
-	return s.http.Serve(listener)
-}
-
-func (s *Server) serveHTTPS() error {
-	var listener net.Listener
-	var err error
-
-	if s.HTTPSAddr == "SocketTLS" {
-		listener = tls.NewListener(getSocketListener(s.HTTPSAddr), s.https.TLSConfig)
-	} else {
-		listener, err = tls.Listen("tcp", s.HTTPSAddr, s.https.TLSConfig)
-		if err != nil {
-			log.Fatal("[zap] unable to create tls listener", err)
-		}
-	}
-
-	log.Println("[zap] https listening at", listener.Addr())
-	return s.https.Serve(listener)
-}
-
 func createHTTPServer() *http.Server {
 	mux := http.NewServeMux()
 	mux.HandleFunc("/", findAppHandler(appHandler))
@@ -130,12 +94,4 @@ func createHTTPSServer() *http.Server {
 	http2.ConfigureServer(server, nil)
 
 	return server
-}
-
-func getSocketListener(socket string) net.Listener {
-	listeners, err := launch.SocketListeners(socket)
-	if err != nil {
-		log.Fatal("[zap] unable to get launchd socket listener", err)
-	}
-	return listeners[0]
 }

--- a/zap/server_darwin.go
+++ b/zap/server_darwin.go
@@ -1,0 +1,51 @@
+package zap
+
+import (
+	"crypto/tls"
+	"log"
+	"net"
+
+	"github.com/puma/puma-dev/dev/launch"
+)
+
+func (s *Server) serveHTTP() error {
+	var listener net.Listener
+	var err error
+
+	if s.HTTPAddr == "Socket" {
+		listener = getSocketListener(s.HTTPAddr)
+	} else {
+		listener, err = net.Listen("tcp", s.HTTPAddr)
+		if err != nil {
+			log.Fatal("[zap] unable to create listener", err)
+		}
+	}
+
+	log.Println("[zap] http listening at", listener.Addr())
+	return s.http.Serve(listener)
+}
+
+func (s *Server) serveHTTPS() error {
+	var listener net.Listener
+	var err error
+
+	if s.HTTPSAddr == "SocketTLS" {
+		listener = tls.NewListener(getSocketListener(s.HTTPSAddr), s.https.TLSConfig)
+	} else {
+		listener, err = tls.Listen("tcp", s.HTTPSAddr, s.https.TLSConfig)
+		if err != nil {
+			log.Fatal("[zap] unable to create tls listener", err)
+		}
+	}
+
+	log.Println("[zap] https listening at", listener.Addr())
+	return s.https.Serve(listener)
+}
+
+func getSocketListener(socket string) net.Listener {
+	listeners, err := launch.SocketListeners(socket)
+	if err != nil {
+		log.Fatal("[zap] unable to get launchd socket listener", err)
+	}
+	return listeners[0]
+}

--- a/zap/server_linux.go
+++ b/zap/server_linux.go
@@ -1,0 +1,33 @@
+package zap
+
+import (
+	"crypto/tls"
+	"log"
+	"net"
+)
+
+func (s *Server) serveHTTP() error {
+	var listener net.Listener
+	var err error
+
+	listener, err = net.Listen("tcp", s.HTTPAddr)
+	if err != nil {
+		log.Fatal("[zap] unable to create listener", err)
+	}
+
+	log.Println("[zap] http listening at", listener.Addr())
+	return s.http.Serve(listener)
+}
+
+func (s *Server) serveHTTPS() error {
+	var listener net.Listener
+	var err error
+
+	listener, err = tls.Listen("tcp", s.HTTPSAddr, s.https.TLSConfig)
+	if err != nil {
+		log.Fatal("[zap] unable to create tls listener", err)
+	}
+
+	log.Println("[zap] https listening at", listener.Addr())
+	return s.https.Serve(listener)
+}

--- a/zap/server_windows.go
+++ b/zap/server_windows.go
@@ -1,0 +1,33 @@
+package zap
+
+import (
+	"crypto/tls"
+	"log"
+	"net"
+)
+
+func (s *Server) serveHTTP() error {
+	var listener net.Listener
+	var err error
+
+	listener, err = net.Listen("tcp", s.HTTPAddr)
+	if err != nil {
+		log.Fatal("[zap] unable to create listener", err)
+	}
+
+	log.Println("[zap] http listening at", listener.Addr())
+	return s.http.Serve(listener)
+}
+
+func (s *Server) serveHTTPS() error {
+	var listener net.Listener
+	var err error
+
+	listener, err = tls.Listen("tcp", s.HTTPSAddr, s.https.TLSConfig)
+	if err != nil {
+		log.Fatal("[zap] unable to create tls listener", err)
+	}
+
+	log.Println("[zap] https listening at", listener.Addr())
+	return s.https.Serve(listener)
+}

